Some small changes to CONTRIBUTING.md (#11930)
ocaml/ocaml: CONTRIBUTING.md

to affect threading, it's probably not your fault.

## Description of the proposed change

### In the merge request interface

The description of the merge request must contain a precise
explanation of the proposed change.

Before going into the implementation details, you should include
a summary of the change, a justification of why it is beneficial, and
a high-level description of the design of the proposed change with
example use cases.

We insist that you should *motivate* the change carefully. Any change
has a cost; for example, it requires reviewing work and risks
introducing regressions, it may make the system larger or more

> complex. Explain what are the benefits that compensate for the costs.

Reviewer: I find this paragraph overly negative, the `insist` reinforced by a `carefully` does give the impression that there some criticism hidden behind the scene.

Would it not be better to focus on the reason why a good motivation is an important of a PR? Something like

> Writing this summary is a good time for introspecting on the design choices and benefits of your PR. Change for the sake of change creates friction with other contributions and may accidentally introduce new bugs. Communicating as clearly as you can the benefits of your PR will reassure and motivate potential reviewers.

or

> Explaining the benefits of your PR plays an important part into motivating other people to reviewing your code. Change for the sake of change creates friction with other contributions and may accidentally introduce new bugs. Taking the time to document your design choice and the benefits brought by your PR will alleviate this fear and motivate people into reviewing your PR.

?

Author: I modified the PR following your suggestion. I wasn't fond of "introspecting on" and of "change for the sake of change" (which I find a fairly negative suggestion, more than what I had written before), so I removed parts and the end result is shorter. What do you think?

explanation of the proposed change.

> Before going in the implementation details, you should include

Reviewer 2: ```suggestion
Before going into the implementation details, you should include
```

a summary of the change, a justification of why it is beneficial, and
a high-level description of the design of the proposed change, with

> example use-cases.

Reviewer 2: ```suggestion
a high-level description of the design of the proposed change with
example use cases.
```

We insist that you should *motivate* the change carefully. Any change
has a cost; for example it requires reviewing work and risks
introducing regressions, many changes make the system larger or more

> complex, etc, Explain what the benefits are that compensate the costs.

Reviewer 2: ```suggestion
has a cost; for example, it requires reviewing work and risks
introducing regressions, it may make the system larger or more
complex. Explain what are the benefits that compensate for the costs.
```
